subsys/bluetooth/controller/hci_driver.c

```diff
@@ -356,7 +356,7 @@ static void recv_thread(void *p1, void *p2, void *p3)
 	ARG_UNUSED(p2);
 	ARG_UNUSED(p3);
 
-	static uint8_t hci_buffer[CONFIG_BT_RX_BUF_LEN];
+	static uint8_t hci_buffer[CONFIG_BT_BUF_ACL_RX_SIZE + 4];
```

Comment:
This HCI buffer is used to store incoming events _or_ ACL data. The buffer therefore needs to be sized accordingly. 
```suggestion
	static uint8_t hci_buffer[MAX(CONFIG_BT_BUF_ACL_RX_SIZE, BT_BUF_EVT_RX_SIZE)];
```

Reply:
BT_BUF_RX_SIZE should be used for this purpose. It handles the maximum of the ACL and Events as well as the different header sizes.